homeassistant/components/mqtt/models.py

```diff
@@ -289,12 +295,51 @@ def async_render_with_possible_json_value(
         return rendered_payload
 
 
+class EntityMonitor:
+    """Monitors entity state changes."""
+
+    __slots__ = ("_entity", "_attributes")
+
+    def __init__(self, entity: Entity) -> None:
+        """Initialize entity monitor."""
+        self._entity: Entity = entity
+        self._attributes: dict[str, Any] = {}
+
+    def track(self, attributes: set[str]) -> None:
+        """Start tracking attributes."""
+        self._attributes = {
+            attribute: getattr(self._entity, attribute, UNDEFINED)
+            for attribute in attributes
+        }
+
+    @property
+    def assume_has_changed(self) -> bool:
+        """Return True if attributes on entity changed or if update is forced.
+
+        Stops tracking attribute changes.
+        """
+        entity = self._entity
+        attributes = self._attributes
+        if not (
+            assume_has_changed := (
+                getattr(entity, "_attr_force_update", False) or not attributes
+            )
+        ):
+            for attribute, last_value in attributes.items():
+                if getattr(entity, attribute, UNDEFINED) != last_value:
+                    assume_has_changed = True
+                    break
+
+        attributes.clear()
+        return assume_has_changed
```

Comment:
```suggestion
            for attribute, last_value in attributes.items():
                if getattr(entity, attribute, UNDEFINED) != last_value:
                    return True

        return False
```

I don't think we need to clear here, so we can return right away as soon as we know its changed

Reply:
The reason to clear is that we can have mulitple attributes that can be received from a share topic. If we do not clear, we will also need to check the older attributes.